File: frontend/src/app/shared/models/car.ts
export interface ICar {
    id: number;
    name: string;
    isAvailable: string;
    rentalPrice: number;
    pictureUrl: string;
    carMaker: string;
    carModel: string;
  }
  
  export interface ICarToCreate {
  name: string;
  isAvailable: string;
  rentalPrice: number;
  pictureUrl: string;
  carMakerId: number;
  carModelId: number;
  }
  
  export class CarFormValues implements ICarToCreate {
  name = '';
  isAvailable = '';
  rentalPrice = 0;
  pictureUrl = '';
  carMakerId!: number;
  carModelId!: number;
  
  constructor(init?: CarFormValues) {
    Object.assign(this, init);
  }
  }

File: Core/Entities/Car.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Core.Entities
{
    public class Car : BaseEntity
    {
        public string Name { get; set; }
        public string IsAvailable { get; set; }
        public decimal RentalPrice { get; set; }
        public string PictureUrl { get; set; }
        public CarMaker CarMaker { get; set; }
        public int CarMakerId { get; set; }
        public CarModel CarModel { get; set; }
        public int CarModelId { get; set; }
    }
}